pkg/kv/store_message.go

```diff
@@ -64,8 +64,66 @@ func (s *StoreMessage) Delete(ctx context.Context, path string) error {
 	return err
 }
 
-// TODO(niro): implement Scan when required
+// Scan returns a prefix iterator which returns entries in deserialized format. Scan in store message implementation is msg specific
+// and therefore must be PREFIX bound
```

Comment:
So why isn't `prefix` being validated?

Reply:
Can you explain what kind of validations you had in mind?
Iterator-wise we are using the PrefixIterator which checks for it

Reply:
prefix is a byte slice and the comparison in PrefixIterator is `!bytes.HasPrefix(entry.Key, b.Prefix)` so it will return false for all keys and iterate over all kv store

Reply:
So it can be unbounded and therefore the comment needs to change, right?